### PR TITLE
small Refactoring for jenkins client

### PR DESCRIPTION
**Pull request by jkhelil**
# Changes

- Build the jenkinsclient from a configuration
- Jenkinsclient is intended to be used elsewhere than in the Baseconfiguration.reconcile 
# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes tests (if functionality changed/added)
- [ ] Includes docs (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/jenkinsci/kubernetes-operator/blob/master/CONTRIBUTING.md#commit-messages)

_See [the contribution guide](https://github.com/jenkinsci/kubernetes-operator/blob/master/CONTRIBUTING.md) for more details._

## Reviewer Notes

If API changes are included, additive changes must be approved by at least two [OWNERS](https://github.com/jenkinsci/kubernetes-operator/blob/master/OWNERS) and backwards incompatible changes must be approved by [more than 50% of the OWNERS](https://github.com/jenkinsci/kubernetes-operator/blob/master/OWNERS).

# Release Notes

```
Describe any user facing changes here, or delete this block.

Examples of user facing changes:
- API changes
- Bug fixes
- Any changes in behavior

```

### CODE REVIEW

File: pkg/controller/jenkins/configuration/base/pod.go [hunk 1]
```diff
@@ -147,7 +147,7 @@ func (r *ReconcileJenkinsBaseConfiguration) ensureJenkinsMasterPod(meta metav1.O
 	}
 
 	// Check if this Pod already exists
-	currentJenkinsMasterPod, err := r.getJenkinsMasterPod()
+	currentJenkinsMasterPod, err := getJenkinsMasterPod(&r.Configuration)
```
Comment (tomaszsek) [inline pkg/controller/jenkins/configuration/base/pod.go:150]:
```suggestion
	currentJenkinsMasterPod, err := r.Configuration.getJenkinsMasterPod()
```

File: pkg/controller/jenkins/configuration/base/reconcile.go [hunk 2]
```diff
@@ -299,7 +273,7 @@ func (r *ReconcileJenkinsBaseConfiguration) compareVolumes(actualPod corev1.Pod)
 }
 
 func (r *ReconcileJenkinsBaseConfiguration) detectJenkinsMasterPodStartingIssues() (stopReconcileLoop bool, err error) {
-	jenkinsMasterPod, err := r.getJenkinsMasterPod()
+	jenkinsMasterPod, err := getJenkinsMasterPod(&r.Configuration)
```
Comment (tomaszsek) [inline pkg/controller/jenkins/configuration/base/reconcile.go:276]:
```suggestion
	jenkinsMasterPod, err := r.Configuration.getJenkinsMasterPod()
```

File: pkg/controller/jenkins/configuration/base/reconcile.go [hunk 3]
```diff
@@ -350,7 +324,7 @@ func (r *ReconcileJenkinsBaseConfiguration) filterEvents(source corev1.EventList
 }
 
 func (r *ReconcileJenkinsBaseConfiguration) waitForJenkins() (reconcile.Result, error) {
-	jenkinsMasterPod, err := r.getJenkinsMasterPod()
+	jenkinsMasterPod, err := getJenkinsMasterPod(&r.Configuration)
```
Comment (tomaszsek) [inline pkg/controller/jenkins/configuration/base/reconcile.go:327]:
```suggestion
	jenkinsMasterPod, err := r.Configuration.getJenkinsMasterPod()
```

File: pkg/controller/jenkins/configuration/base/reconcile.go [hunk 4]
```diff
@@ -507,14 +381,14 @@ func (r *ReconcileJenkinsBaseConfiguration) ensureBaseConfiguration(jenkinsClien
 }
 
 func (r *ReconcileJenkinsBaseConfiguration) waitUntilCreateJenkinsMasterPod() (currentJenkinsMasterPod *corev1.Pod, err error) {
-	currentJenkinsMasterPod, err = r.getJenkinsMasterPod()
+	currentJenkinsMasterPod, err = getJenkinsMasterPod(&r.Configuration)
```
Comment (tomaszsek) [inline pkg/controller/jenkins/configuration/base/reconcile.go:384]:
```suggestion
	currentJenkinsMasterPod, err = r.Configuration.getJenkinsMasterPod()
```

File: pkg/controller/jenkins/configuration/base/reconcile.go [hunk 5]
```diff
@@ -507,14 +381,14 @@ func (r *ReconcileJenkinsBaseConfiguration) ensureBaseConfiguration(jenkinsClien
...
 	for {
 		if err != nil && !apierrors.IsNotFound(err) {
 			return nil, stackerr.WithStack(err)
 		} else if err == nil {
 			break
 		}
-		currentJenkinsMasterPod, err = r.getJenkinsMasterPod()
+		currentJenkinsMasterPod, err = getJenkinsMasterPod(&r.Configuration)
```
Comment (tomaszsek) [inline pkg/controller/jenkins/configuration/base/reconcile.go:391]:
```suggestion
		currentJenkinsMasterPod, err = r.Configuration.getJenkinsMasterPod()
```

File: pkg/controller/jenkins/configuration/configuration.go [hunk 6]
```diff
@@ -3,12 +3,15 @@ package configuration
...
 	"context"
+	"strings"
+	"time"
 
 	"github.com/jenkinsci/kubernetes-operator/pkg/apis/jenkins/v1alpha2"
 	"github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/configuration/base/resources"
 	"github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/notifications/event"
 	"github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/notifications/reason"
 
+	jenkinsclient "github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/client"
```
Comment (tomaszsek) [inline pkg/controller/jenkins/configuration/configuration.go:14]:
```suggestion
	jenkinsclient "github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/client"

```

File: pkg/controller/jenkins/configuration/configuration_test.go [hunk 7]
```diff
@@ -0,0 +1,144 @@
+package configuration
+
+import (
+	"testing"
+	
+	"github.com/jenkinsci/kubernetes-operator/pkg/apis/jenkins/v1alpha2"
```
Comment (tomaszsek) [inline pkg/controller/jenkins/configuration/configuration_test.go:6]:
```suggestion
	"github.com/jenkinsci/kubernetes-operator/pkg/apis/jenkins/v1alpha2"

```

File: pkg/controller/jenkins/configuration/base/pod.go (hunk 1 above)
Comment (jkhelil) [reply pkg/controller/jenkins/configuration/base/pod.go]: thank you for spotting this, I missed it actually after reworking the PR. Fixed now

File: pkg/controller/jenkins/configuration/configuration_test.go (hunk 7 above)
Comment (jkhelil) [reply pkg/controller/jenkins/configuration/configuration_test.go:6]: Why adding the NL here ?

Comment (tomaszsek) [reply pkg/controller/jenkins/configuration/configuration_test.go:6]: We have a pattern:
```
<standard-imports>

<application-imports>

<other-imports>
```

**Pull request merged**